client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java

```diff
@@ -2528,6 +2529,11 @@ public void testGetTrainedModelsStats() throws Exception {
                     .collect(Collectors.toList()),
                 containsInAnyOrder(modelIdPrefix + 1, modelIdPrefix + 2));
         }
+        highLevelClient().ingest().deletePipeline(new DeletePipelineRequest("regression-stats-pipeline"), RequestOptions.DEFAULT);
+        assertBusy(() -> {
+            assertTrue(indexExists(".ml-stats-000001"));
+            ensureGreen(".ml-stats-*");
+        });
```

Comment:
This stinks, but it allows the stats to be persisted and makes sure the stats index exists before continuing. This is a slight pain as the stats persistence for trained models is asynchronous.